phoenix-core/src/main/java/org/apache/phoenix/schema/task/Task.java

```diff
@@ -230,7 +232,20 @@ public static void deleteTask(PhoenixConnection conn, PTable.TaskType taskType,
                 taskQuery += " AND " + PhoenixDatabaseMetaData.TASK_DATA + " LIKE '%" + indexName + "%'";
             }
 
-        return populateTasks(connection, taskQuery);
+            List<TaskRecord> taskRecords = populateTasks(connection, taskQuery);
+            List<TaskRecord> result = new ArrayList<TaskRecord>();
+            if (ts != null) {
```

Comment:
Let me try again I accidentally commented in test code.  Here we have a simple logic change where we query against a property.  This is tested in the ITs but it is good to test with a small simple unit test.  However, it might pose a bit more difficult to test if there isn't a boundary between the data layer and the logic.  This may inform how to shape our code.